### PR TITLE
[8.0] Keeping track of network wait time to client

### PR DESCRIPTION
**Pull request by riverszhang89**
This patch keeps track of how much time we are blocked on network IO to client, for instance:

```
11/06 18:57:56: LONG REQUEST 9562 msec ...
11/06 18:57:56:   sql=select * from jobs, sql=select * from jobs, ncols=10
11/06 18:57:56:   netwaitms=9148, rowcount=48445, cost=48454.000000
```

### CODE REVIEW

**Review by mponomar — COMMENTED** [review]
This is great, thank you.  Maybe add to eventlog so we have this data per query?

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/559 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>truncatesc_offline_generated<br>dbcreate</p>

File: net/sqlwriter.c [hunk 1]
```diff
@@ -177,19 +204,26 @@ static void sql_flush_cb(int fd, short what, void *arg)
...
-        writer->bad = 1;
-        event_del(writer->flush_ev);
+    } else if (n <= 0) {
+        if (errno == EAGAIN || errno == EWOULDBLOCK) {
+            update_writer_state(writer, WRITE_BLOCKED);
+        } else {
+            writer->bad = 1;
+            event_del(writer->flush_ev);
+            update_writer_state(writer, WRITE_FAILED);
+        }
```
Comment (morgando) [inline net/sqlwriter.c:218]: Nitpick: Maybe the other state changes to `writer` that happen here could also go into `update_writer_state`? (like setting `writer->bad` and `writer->sent_at`)

File: tests/reqlog_netwait.test/runit [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+mach=`cdb2sql --tabs ${CDB2_OPTIONS} $dbnm default "SELECT comdb2_host()"`
+
+if [ "$mach" = "" ]; then
+   echo could not retrieve hostname >&2
+   exit 1
+fi
+
+# Redirect longreqs to standard out
+cdb2sql ${CDB2_OPTIONS} --host $mach $dbnm "EXEC PROCEDURE sys.cmd.send('reql longreqfile <stdout>')" >/dev/null
```
Comment (morgando) [inline tests/reqlog_netwait.test/runit:17]: TIL!

**Review by morgando — APPROVED** [review]
LGTM!

**Pull request merged**